numpy/core/src/multiarray/arraytypes.c.src

```diff
@@ -3803,17 +3803,23 @@ BOOL_dot(char *ip1, npy_intp is1, char *ip2, npy_intp is2, char *op, npy_intp n,
     *((npy_bool *)op) = tmp;
 }
 
+/*
+ * `dot` does not make sense for times, for DATETIME it never worked.
+ *  For timedelta it does/did , but should probably also just be removed.
+ */
+#define DATETIME_dot NULL
+
```

Comment:
This is a bit of a madness, but `np.result_type(bool, "M8")` never worked, and that prevented us (correctly) from even trying to get a datetime dot, vdot, or correlate.

Now it doesn't make any sense for timedelta as well (nor does it really work right, since units are always dropped), but not doing that here.  This just "preserves" breaking things reliably, becuase we definitely cannot call the functions if they are not defined ;).